src/query/storage/m3/storage.go

```diff
@@ -310,7 +310,14 @@ func (s *m3storage) fetchCompressed(
 
 	matchOpts := s.opts.SeriesConsolidationMatchOptions()
 	tagOpts := s.opts.TagOptions()
-	result := consolidators.NewMultiFetchResult(fanout, pools, matchOpts, tagOpts)
+	limitOpts := consolidators.LimitOptions{
+		Limit: options.SeriesLimit,
+		// Piggy back on the new InstanceMultiple option to enable checking require exhaustive. This preserves the
+		// existing buggy behavior of the coordinators not requiring exhaustive. Once InstanceMultiple is enabled by
+		// default, this can be removed.
+		RequireExhaustive: queryOptions.InstanceMultiple > 0 && options.RequireExhaustive,
```

Comment:
not really following why this will be set to false if `queryOptions.InstanceMultiple == 0`

Reply:
to preserve backwards compatibility with the buggy logic to not check `RequireExhaustive` in the coord. I'm piggy backing on the rollout of `InstanceMultiple` to also enable checking `RequireExhaustive` in the coord. Will remove `queryOptions.InstanceMultiple > 0` once we verify this all works.